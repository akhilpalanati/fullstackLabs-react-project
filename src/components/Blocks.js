import React from "react";
import PropTypes from "prop-types";

import {CircularProgress, Typography, makeStyles} from "@material-ui/core";
import colors from "../constants/colors";

const Blocks = ({ data= [], loading}) => {
  function indexValue(index) {
    if (index <= 9) {
      return `00${index}`;
    }
    if (index >= 10 || index <= 99) {
      return `0${index}`;
    }
    return index;
  }
  const classes = useStyles();
  return (
    <div>
      {
        loading ? <CircularProgress /> :
          data.map(({attributes: {data, index}}) => (
            <div className={classes.contentElement} key={index}>
              <Typography className={classes.indexValue}>{indexValue(index)}</Typography>
              <Typography>{data}</Typography>
            </div>
        ))

      }
    </div>
  )
}

const useStyles = makeStyles( () => ({
  contentElement: {
    backgroundColor: colors.contentBackground,
    padding: '10px',
    margin: '5px'

  },
  indexValue: {
    fontSize: '12px',
   color: colors.primary
  }
  }))

Blocks.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      attributes: PropTypes.shape({
        index: PropTypes.string,
        data: PropTypes.string
      })
    })
  ).isRequired,
  loading: PropTypes.bool.isRequired
};

export default Blocks;
