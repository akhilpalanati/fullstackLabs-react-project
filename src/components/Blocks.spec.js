import React from 'react';
import  Blocks from './Blocks';
import { create } from "react-test-renderer";

describe("<Blocks/>", () => {
  const data = [{
    attributes: {
      index:"1",
      data: "The human Torch"
    }},{
    attributes: {
      index:"2",
      data: "The human Torch"
    }

  }]

  it("should match snapshot", () => {
    const component = create(
      <Blocks data ={data} loading ={false}/>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
})
