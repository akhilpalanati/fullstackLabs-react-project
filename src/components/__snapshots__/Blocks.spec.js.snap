// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks/> should match snapshot 1`] = `
<div>
  <div
    className="makeStyles-contentElement-1"
  >
    <p
      className="MuiTypography-root makeStyles-indexValue-2 MuiTypography-body1"
    >
      001
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      The human Torch
    </p>
  </div>
  <div
    className="makeStyles-contentElement-1"
  >
    <p
      className="MuiTypography-root makeStyles-indexValue-2 MuiTypography-body1"
    >
      002
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      The human Torch
    </p>
  </div>
</div>
`;
